### PR TITLE
Feature: key prefix option

### PR DESCRIPTION
**Pull request by mdjastrzebski**
Adds optional `keyPrefix` parameter to `getFixedT` function in order to allows for improved developer experience when querying deeply nested keys with common prefix.

```
i18next.addResource('fr', 'translation', 'very.deeply.nested.key', 'ici!');
const t = i18next.getFixedT('fr', null, 'very.deeply.nested');
const test = t('key'); // "ici!"
```

Original GH issue (from react-i18next): https://github.com/i18next/react-i18next/issues/1359 

#### Checklist

- [x] only relevant code is changed (make a diff before you submit the PR)
- [x] run tests `npm run test`
- [x] tests are included

#### Checklist (for documentation change)

- [ ] only relevant documentation part is changed (make a diff before you submit the PR)
- [ ] motivation/reason is provided

### CODE REVIEW

**Comment by mdjastrzebski** [comment]
I would like to also update the docs, I'm not sure how to do it, as I haven't found docs sources in this repo.

**Comment by adrai** [comment]
The docs are here: https://github.com/i18next/i18next-gitbook/blob/master/overview/api.md#getfixedt

**Comment by adrai** [comment]
beside the typescript signature, it lgtm

File: index.d.ts [hunk 1]
```diff
@@ -940,9 +940,15 @@ export interface i18n {
...
+   * Accepts optional keyPrefix that will be automatically applied to returned t function.
    */
-  getFixedT(lng: string | readonly string[], ns?: string | readonly string[]): TFunction;
-  getFixedT(lng: null, ns: string | readonly string[]): TFunction;
+  getFixedT(
+    lng: string | readonly string[],
+    ns?: string | readonly string[],
+    keyPrefix?: string,
+  ): TFunction;
+  getFixedT(lng: null, ns: string | readonly string[], keyPrefix?: string): TFunction;
```
[the remark this answers is not in the document]
Comment (mdjastrzebski) [reply index.d.ts]: I've updated second signature to allow `null` value for `ns` argument. This should have the same effect as creating third signature `getFixedT(lng: null, ns: null, keyPrefix?: string): TFunction;`. @adrai wdyt?

**Comment by adrai** [comment]
Thank you. v20.6.0 has just been released.

**Pull request merged**